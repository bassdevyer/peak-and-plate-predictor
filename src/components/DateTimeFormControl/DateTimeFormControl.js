import React from 'react'
import './style.css'
import { FormControl } from 'react-bootstrap'
import PropTypes from 'prop-types'

const propTypes = {
    id: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    type: PropTypes.string,
    value: PropTypes.string.isRequired,

  },
  defaultProps = {
    type: 'datetime-local',
  }

const DateTimeFormControl = (props) => {
  const {
    id,
    onChange,
    type,
    value,
  } = props

  return (
    <FormControl id={id}
                 onChange={onChange}
                 type={type}
                 value={value}/>
  )
}

DateTimeFormControl.propTypes = propTypes
DateTimeFormControl.defaultProps = defaultProps

export default DateTimeFormControl