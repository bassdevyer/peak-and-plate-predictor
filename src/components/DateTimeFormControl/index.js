import DateTimeFormControl from './DateTimeFormControl'

export default DateTimeFormControl