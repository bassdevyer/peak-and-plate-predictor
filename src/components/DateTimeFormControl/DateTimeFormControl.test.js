import React from 'react'
import DateTimeFormControl from '../DateTimeFormControl'
import { create } from 'react-test-renderer'
import ReactDOM from 'react-dom'

describe('Component Mounting', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<DateTimeFormControl/>, div)
  })
})

describe('Component default rendering', () => {
  test('Renders default component', () => {
    const component = create(
      <DateTimeFormControl/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
