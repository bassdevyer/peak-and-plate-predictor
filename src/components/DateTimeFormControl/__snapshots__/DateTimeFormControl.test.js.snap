// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default rendering Renders default component 1`] = `
<input
  className="form-control"
  type="datetime-local"
/>
`;
